book/src/dev/rfcs/0012-value-pools.md

```diff
@@ -0,0 +1,437 @@
+
+As defined in [ZIP-209], the Sapling chain value pool balance for a given block chain is the negation of the sum of all valueBalanceSapling field values for transactions in the block chain.
+
+Consensus rule: If the Sapling chain value pool balance would become negative in the block chain created as a result of accepting a block , then all nodes MUST reject the block as invalid.
+
+https://zips.z.cash/protocol/protocol.pdf#saplingbalance
+
+### Orchard Chain Value Pool
+
+Orchard introduces Action transfers, each of which can optionally perform a spend, and optionally perform an output. Similarly to Sapling, the net value of Orchard spends minus outputs in a transaction is called the Orchard balancing value, measured in zatoshi as a signed integer vbalanceOrchard.
+
+vbalanceOrchard is encoded in a transaction as the field valueBalanceOrchard. If a transaction has no Action descriptions, vbalanceOrchard is implicitly zero. Transaction fields are described in § 7.1 ‘Transaction Encoding and Consensus’ on p. 116.
+
+A positive Orchard balancing value takes value from the Orchard transaction value pool and adds it to the transpar- ent transaction value pool . A negative Orchard balancing value does the reverse. As a result, positive vbalanceOrchard is treated like an input to the transparent transaction value pool , whereas negative vbalanceOrchard is treated like an output from that pool.
+
+Similarly to the Sapling chain value pool balance defined in [ZIP-209], the Orchard chain value pool balance for a given block chain is the negation of the sum of all valueBalanceOrchard field values for transactions in the block chain.
+
+Consensus rule: If the Orchard chain value pool balance would become negative in the block chain created as a result of accepting a block , then all nodes MUST reject the block as invalid.
+
+https://zips.z.cash/protocol/protocol.pdf#orchardbalance
+
+## Proposed Implementation
+
+### Create a new `ValueBalance` type
+
+- Code will be located in a new file: `zebra-chain/src/value_balance.rs`.
+- Supported operators apply to all the `Amount`s inside the type: `+`, `-`, `+=`, `-=`, `sum()`.
+- We will use `Default` to represent a totally empty `ValueBalance`, this is the state of all pools at the genesis block.
+
+```rust
+struct ValueBalance<C> {
+    transparent: Amount<C>,
+    sprout: Amount<C>,
+    sapling: Amount<C>,
+    orchard: Amount<C>,
+}
+
+impl ValueBalance {
+    /// Consensus rule: The remaining value in the transparent transaction value pool MUST be nonnegative.
+    ///
+    /// This rule applies to Block and Mempool transactions
+    fn remaining_transparent_value(&self) -> Result<Amount<NonNegative>,Err> {
+        // This rule checks the sum of the transparent, sprout, sapling, and orchard value balances in a transaction, not `ValueBalance.transparent`.
+        [self.transparent, self.sprout, self.sapling, self.orchard].sum()
```

Comment:
See comment above - I think this would be `self.transparent - [self.sprout + self.sapling + self.orchard].sum()` or otherwise change the interpretation of `self.orchard` to be `valueBalanceOrchard`

Reply:
Thanks, i added the suggested code at https://github.com/ZcashFoundation/zebra/pull/2430/commits/996e86531fc8e85b8393dea0e236a788d095db0d